src/pickup.cpp

```diff
@@ -271,12 +272,58 @@ static bool select_autopickup_items( std::vector<item> &here, std::vector<pickup
     return bFoundSomething;
 }
 
+enum pickup_answer :int {
+    CANCEL = -1,
+    WIELD,
+    WEAR,
+    SPILL,
+    STASH,
+    NUM_ANSWERS
+};
+
+pickup_answer handle_problematic_pickup( const item &it, bool &offered_swap, const std::string &explain )
+{
+    if( !g->u.is_armed() && !g->u.keep_hands_free ) {
```

Comment:
Extracting `g->u` into a local variable `player &u = g->u;` makes the code a bit easier to read. But more importantly, it allows to make that reference a parameter when needed later.